### PR TITLE
Adds a new gas reaction, plasmic fusion.

### PR DESCRIPTION
**Pull request by as334**
When heated to high temperatures with carbon dioxide present, plasma undergoes a powerful fusion reaction that produces shitloads of energy, as well as some nitrogen and oxygen fusion byproducts, however the reaction is reduced in efficiency by the presence of non-plasma/carbon dioxide contaminants, so a proper filter setup must be maintained.

:cl: as334
rscadd: New research has shown that when when plasma reaches a high enough temperature in the presence of a carbon catalyst, it undergoes a violent and powerful fusion reaction.
rscadd: Adds a new gas reaction, fusion. Heat up plasma and carbon dioxide in order to produce large amounts of energy and heat.
/:cl:

### CODE REVIEW

**Comment by Dorsisdwarf** [comment]
Oooh, time for a fusion engine

**Comment by Knigga** [comment]
:+1:  atmos always needs more things to do

File: code/datums/gas_mixture.dm [hunk 1]
```diff
@@ -120,6 +120,32 @@ What are the archived variables for?
...
+				//world << "pre [temperature, [toxins], [carbon_dioxide]
+				var/old_heat_capacity = heat_capacity()
+				var/carbon_efficency = min(toxins/carbon_dioxide,MAX_CARBON_EFFICENCY)
+				var/reaction_energy = thermal_energy()
+				var/moles_impurities = total_moles()-(plasma+carbon_dioxide)
+
+				var/plasma_fused = (PLASMA_FUSED_COEFFICENT*carbon_effiency)*(temperature/PLASMA_BINDING_ENERGY)
+				var/carbon_catalyzed = (CARBON_CATALYST_COEFFICENT*carbon_efficency)*(temperature/PLASMA_BINDING_ENERGY)
+				var/oxygen_added+=carbon_catalyzed
+				var/nitrogen_added+=plasma_fused-oxygen_added
```
Comment (duncathan) [inline code/datums/gas_mixture.dm]: Are you sure these should be `+=`?

File: code/datums/gas_mixture.dm [hunk 2]
```diff
@@ -120,6 +120,32 @@ What are the archived variables for?
...
+
+				reaction_energy += ((carbon_efficency*plasma)/((moles_impurities/carbon_efficency)+2)*10)+((plasma_fused/(moles_impurities/carbon_efficency))*PLASMA_BINDING_ENERGY)
+				plasma-=plasma_fused
+				carbon-=carbon_catalyzed
```
Comment (duncathan) [inline code/datums/gas_mixture.dm]: >catalyst
>gets consumed in the reaction

**Comment by duncathan** [comment]
Code is pretty okay for gas mixtures :+1:

File: code/datums/gas_mixture.dm (hunk 1 above)
Comment (as334) [inline code/datums/gas_mixture.dm]: Oh yeah, i'll change that.

File: code/datums/gas_mixture.dm (hunk 2 above)
Comment (as334) [inline code/datums/gas_mixture.dm]: It's actually the Carbon itself that's the catalyst, the Oxygen part is what's getting removed, however we don't have a pure carbon gas or graphite, so i did the best i could. notice how the carbon gets converted into 02?

File: code/datums/gas_mixture.dm [hunk 3]
```diff
@@ -120,6 +120,32 @@ What are the archived variables for?
 					temperature -= (reaction_rate*20000)/heat_capacity()
 
 					reacting = 1
+		if(thermal_energy > PLASMA_BINDING_ENERGY)
```
Comment (duncathan) [inline code/datums/gas_mixture.dm]: `thermal_energy()`

File: code/datums/gas_mixture.dm [hunk 4]
```diff
@@ -120,6 +120,32 @@ What are the archived variables for?
...
+				var/reaction_energy = thermal_energy()
+				var/moles_impurities = total_moles()-(plasma+carbon_dioxide)
+
+				var/plasma_fused = (PLASMA_FUSED_COEFFICENT*carbon_effiency)*(temperature/PLASMA_BINDING_ENERGY)
+				var/carbon_catalyzed = (CARBON_CATALYST_COEFFICENT*carbon_efficency)*(temperature/PLASMA_BINDING_ENERGY)
+				var/oxygen_added+=carbon_catalyzed
+				var/nitrogen_added+=plasma_fused-oxygen_added
+
+				reaction_energy += ((carbon_efficency*plasma)/((moles_impurities/carbon_efficency)+2)*10)+((plasma_fused/(moles_impurities/carbon_efficency))*PLASMA_BINDING_ENERGY)
+				plasma-=plasma_fused
```
Comment (duncathan) [inline code/datums/gas_mixture.dm]: `toxins`

File: code/datums/gas_mixture.dm (hunk 2 above)
Comment (duncathan) [inline code/datums/gas_mixture.dm]: Also this should be `carbon_dioxide`

File: code/datums/gas_mixture.dm (hunk 4 above)
Comment (as334) [inline code/datums/gas_mixture.dm]: Yeah i fixed all that jazz.

File: code/datums/gas_mixture.dm (hunk 2 above)
Comment (bgobandit) [inline code/datums/gas_mixture.dm]: what the fuck is that

**Comment by Jalleo** [comment]
What will this mean for the TEG/Turbine in terms of viability?

**Comment by as334** [comment]
It requires quite a bit of setup, and it'd be hard to setup for the turbine due to that using tile atmos, but for TEG a well made fusion reactor could probably be a pretty solid source of power.

**Comment by CosmicScientist** [comment]
But I thought fusion had one byproduct due to its very nature? (Not saying this is bad, just thinking about the science).

**Comment by Dorsisdwarf** [comment]
Oxygen is produced by two plasma atoms and a carbon atom fusing,  nitrogen is produced by three plasma and two carbons, Tadaaaa.

**Comment by duncathan** [comment]
It's something like this:
Plasma + C = N2
And consequently
Plasma + CO2 = N2 + O2

**Comment by as334** [comment]
Basically. I wish we had more than 2 gases that i could have as waste products so it would make sense. And if we get some more expandable options I will probably add some more byproducts.

**Comment by WJohn** [comment]
What about the toxins chamber? You spend a while burning plasma at a very high temp and pressure, and it's left with tons of CO2. It's not like you can use all that power for anything, but it'd be nice if you somehow could.

**Comment by as334** [comment]
the impurities would be a major issue in that case, and it would just basically be a really hot but slowly burning fire, since the main source of energy there would just be the introduction of the new oxygen.

**Comment by Shadowlight213** [comment]
What does this mean for bombs though? Does this mean 5 minute max caps again?

**Comment by as334** [comment]
Not a chance. I don't think it's possible to use this for bombs at the moment, though it will be in the future, and it would take way more than 5 minutes to even heat up gas to fusion point. So yeah, no real concerns on that front.

**Comment by Jazaen** [comment]
So this means you CAN'T turn corridors into fusion furnaces during malf?

**Comment by kevinz000** [comment]
DO IT!!

**Comment by KorPhaeron** [comment]
This broke like a dozen things

**Comment by Jalleo** [comment]
Mistakes were going to occur this can get fixed in time.

**Comment by as334** [comment]
It broke one thing that was already broken and this just  revealed it, don't exaggerate

**Pull request merged**